gradio/chat_interface.py

```diff
@@ -304,6 +307,9 @@ def _setup_events(self) -> None:
                 [self.saved_input, self.chatbot_state] + self.additional_inputs,
                 [self.chatbot, self.chatbot_state],
                 api_name=False,
+                concurrency_limit=cast(
```

Comment:
Why do we need this cast?

Reply:
Because otherwise pyright will complain because it forgets that the type is Literal['default'] and uses `str` instead.